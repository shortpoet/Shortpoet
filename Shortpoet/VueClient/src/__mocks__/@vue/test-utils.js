// export default {
//   mount: jest.fn(() => {
//     console.log("mount mock")
//   })
// }

const test = jest.genMockFromModule('@vue/test-utils')
const component = {}
const mountOptions = {
  propsData: {},
  localVue: {},
mocks: {},
store: {},
context: {}, // only for functional components
router: {},
computed: {},
stubs: {},
slots: {},
scopedSlots: {},
attrs: {},
attachToDocument: false,
attachTo: null, //HTML Element or String
listeners: {},
parentComponent: {},
provide: {}
}

test.mount = jest.fn(() => {
  return {
    component,
    mountOptions
  }
})

module.exports = test