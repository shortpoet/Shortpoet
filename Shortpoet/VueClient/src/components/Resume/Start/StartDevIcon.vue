<template>
  <div class="">
    <i v-if="!isImage" :class="source"></i>
    <img v-else :class="imgClass" :src="source"/>
    <span v-if="name" class="devicon-name">{{ name }}</span>
  </div>
</template>

<script>

export default {
  name: 'StartDevIcon',
  props: {
    source: {
      type: String
    },
    name: {
      type: String,
      default: null
    }
  },
  components: {
  },
  data () {
    return {
    }
  },
  computed: {
    isImage () {
      return this.source.match(/svg|jpg|png|jpeg/g) !== null
    },
    imgClass () {
      if (this.source.includes('vue')) {
        return 'vue-devicon'
      } else if (this.source.includes('d3')) {
        return 'd3-devicon'
      } else {
        return 'tableau-devicon'
      }
    }
  },
  methods: {
  },
  mounted () {
  }
}
</script>
