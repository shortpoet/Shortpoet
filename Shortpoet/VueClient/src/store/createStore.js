// import VueCookies from 'vue-cookies'
import { state, getters, mutations, actions, modules } from './index'

// import StoreResume from '@/store/modules/StoreResume'

import { createStoreResumeMocks } from '@/store/modules/resume/createStoreResume'

const createStore = () => {
  return {
    state: state,
    getters: getters,
    mutations: mutations,
    actions: actions,
    modules: modules
  }
}

const createMocks = (custom = {  state: {}, getters: {}, mutations: {}, actions: {}, modules: {} }) => {
  const mockState = Object.assign({}, state, custom.state)
  const mockGetters = Object.assign({}, getters, custom.getters)
  const mockMutations = Object.assign({}, mutations, custom.mutations)
  const mockActions = Object.assign({}, actions, custom.actions)
  const mockModules = Object.assign({}, modules, custom.modules)

  /* istanbul ignore next */
  // const createStoreResumeMocks = (custom = { getters: {}, mutations: {}, actions: {}, state: {} }) => {
  //   return StoreResume.createStoreResumeMocks(custom)
  // } 

  return  {  
    state: mockState,
    getters: mockGetters,
    mutations: mockMutations,
    actions: mockActions,
    modules: mockModules,
    createStoreResumeMocks: createStoreResumeMocks
  }
}

export default {
  createStore,
  createMocks
}
