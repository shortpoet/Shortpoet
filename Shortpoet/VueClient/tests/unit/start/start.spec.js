import { factoryShallow } from '../test.utils'
import { mount, shallowMount, createLocalVue } from '@vue/test-utils'

import Start from '@/views/Start'
import createStore from '@/store'

import Vuex from 'vuex'
import PortalVue from 'portal-vue'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(PortalVue)

describe('Start.vue', () => {
  it('matches snapshot', () => {
    const getResumeLoaded = true

    // const propsData = { 
    //   interests: sampleInterests 
    // }

    const computed = {
      // getResumeLoaded: () => true
    }

    const mocks = {
      $: jest.fn(() => {
        return {
          click: jest.fn(),
          scrollspy: jest.fn()
        }
      }),
      dispatch: jest.fn(),

      // getResumeLoaded: jest.fn(() => getResumeLoaded)
    }

    // const wrapper = factoryShallow(Start, {mocks: mocks})
    // const wrapper = factoryShallow(Start, {})
    // const wrapper = mount(Start, {mocks: mocks})
    // jest.mock('$')

    const mockStoreResume = createStore.createMocks().createStoreResumeMocks({ mutations: { 'SET_RESUME_RAW': jest.fn() } })
    console.log(mockStoreResume.mutations['SET_RESUME_RAW']())
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})
    const store = new Vuex.Store(storeConfig)
    const wrapper = shallowMount(Start, {localVue, mocks, store, computed})
    expect(wrapper.html()).toMatchSnapshot()
  })
})
