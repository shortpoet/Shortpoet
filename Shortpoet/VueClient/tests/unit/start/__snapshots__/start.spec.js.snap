// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start.vue matches hard resume snapshot 1`] = `
<div id="resume-anchor" class="main-wrapper">
  <nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Carlos Soriano - Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
    <div id="navbarSupportedContent" class="collapse navbar-collapse">
      <ul class="navbar-nav">
        <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
        <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
        <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
        <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
        <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
        <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
      </ul>
    </div>
  </nav>
  <div class="container-fluid p-0">
    <section id="about" class="resume-section p-3 px-lg-5 d-flex d-column">
      <div class="my-auto">
        <h1 class="mb-0">CARLOS
          <span class="text-primary">SORIANO</span></h1>
        <div class="ml-2 d-inline-flex flex-column subheading">
          <div class="d-inline-flex flex-column"><span id="resume-flags" class="not-name">ðŸ‡ªðŸ‡¸ ðŸ‡µðŸ‡­ ðŸ‡ºðŸ‡¸ ðŸ‡¨ðŸ‡³ ðŸ‡«ðŸ‡· ðŸ‡®ðŸ‡¹ ðŸ‡©ðŸ‡ª ðŸ‡§ðŸ‡·</span> <a href="mailto:soriano.carlitos@gmail.com">soriano.carlitos@gmail.com</a> <span id="resume-address">Currently in St. Louis, MO</span></div>
          <div class="d-flex flex-column"><span id="resume-visas" class="not-name">Valid Work Visas: USA, EU, UK</span></div>
        </div>
      </div>
    </section>
    <section id="start-socials" class="resume-section pl-3 pl-lg-5 py-1 d-flex d-column">
      <div class="my-auto">
        <ul class="list-inline list-social-icons mb-0 mt-0">
          <li class="social-item list-inline-item " style="font-family: FontAwesome;"><a href="https://github.com/shortpoet"><span class="fa-stack fa-lg"><i class="fa fa-circle fa-stack-2x"></i> <i class="fa fa-github fa-stack-1x fa-inverse"></i></span></a></li>
          <li class="social-item list-inline-item " style="font-family: FontAwesome;"><a href="https://www.linkedin.com/in/carlos-soriano-49aaa97/"><span class="fa-stack fa-lg"><i class="fa fa-circle fa-stack-2x"></i> <i class="fa fa-linkedin fa-stack-1x fa-inverse"></i></span></a></li>
          <li class="social-item list-inline-item " style="font-family: FontAwesome;"><a href="https://www.instagram.com/shortpoet/"><span class="fa-stack fa-lg"><i class="fa fa-circle fa-stack-2x"></i> <i class="fa fa-instagram fa-stack-1x fa-inverse"></i></span></a></li>
          <li class="social-item list-inline-item " style="font-family: FontAwesome;"><a href="https://twitter.com/shortpoet3"><span class="fa-stack fa-lg"><i class="fa fa-circle fa-stack-2x"></i> <i class="fa fa-twitter fa-stack-1x fa-inverse"></i></span></a></li>
          <li class="social-item list-inline-item " style="font-family: FontAwesome;"><a href="http://localhost/pdf"><span class="fa-stack fa-lg"><i class="fa fa-circle fa-stack-2x"></i> <i class="far fa-file-pdf-o fa-stack-1x fa-inverse"></i></span></a></li>
        </ul>
      </div>
    </section>
    <section id="skills" class="resume-section pl-3 p-lg-5 d-flex flex-column">
      <div class="my-auto">
        <h2 class="mb-4 mb-md-3">Skills</h2>
        <div class="subheading mb-3">
          Programming Languages &amp; Tools
        </div>
        <ul class="list-devicons list-inline list-icons">
          <li class="list-inline-item">
            <div><i class="devicons devicons-visualstudio"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-git"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="fa fa-cogs"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-linux"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-dotnet"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-javascript"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class=""></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-python"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-database"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-msql_server"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-sqllite"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-mysql"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-mongodb"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-html5"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-css3"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-bootstrap"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="devicons devicons-sass"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class=""></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class=""></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="fa fa-beer"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="fa fa-user-ninja"></i>
              <!---->
            </div>
          </li>
          <li class="list-inline-item">
            <div><i class="fa fa-magic"></i>
              <!---->
            </div>
          </li>
        </ul>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Markdown
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              HTML
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               CSS
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               YAML
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               MD
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               SCSS
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Miscellaneous
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              Cooking
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Gardening
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               YogaWorks Teacher Traning
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Cicerone Beer Server Certification
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Adobe Creative Suite
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Data Vizualization
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              D3
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Tableau
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Document Db
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              mongoDB
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Pymongo
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Dotnet
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              Framework 4.x
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Core
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Entity Framework
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Razor
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               MVC
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Python
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              Pandas
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Matplotlib
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Flask
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Selenium
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Sklearn
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Javascript
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              Vue
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Leafly
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Plotly
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               c3
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Workflow
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              Mobile-First &amp; Responsive Design
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Cross Browser Testing &amp; Debugging
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               CI/CD
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Agile Development &amp; Scrum
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Visual Studio / VSCode
            </span></li>
          </ul>
        </div>
        <div class="skill-grid">
          <div class="subheading mb-3">
            Relational Db
          </div>
          <ul class="fa-ul mb-3 skill-list">
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
              msSQL
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               mySQL
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               Entity Framework
            </span></li>
            <li><span class="skill"><i class="skill-icon fa fa-check"></i>
               SQLAlchemy
            </span></li>
          </ul>
        </div>
      </div>
    </section>
    <section id="awards" class="resume-section p-3 p-lg-5 d-flex flex-column">
      <div class="my-auto">
        <h2 class="mb-4 mb-md-3">Natural Languages</h2>
        <ul class="fa-ul mb-0">
          <li><i class="proficient language-icon fa fa-flag fa-li"></i> <span class="mb-5">Italian (Proficient) </span></li>
          <li><i class="native language-icon fa fa-flag fa-li"></i> <span class="mb-5">English, Spanish, Tagalog (Native) </span></li>
          <li><i class="fluent language-icon fa fa-flag fa-li"></i> <span class="mb-5">Mandarin Chinese, French (Fluent) </span></li>
          <li><i class="workingKnowledge language-icon fa fa-flag fa-li"></i> <span class="mb-5">German, Portuguese (Working Knowledge) </span></li>
          <li><i class="award-icon fa fa-trophy fa-li"></i>
            Certified Interpreter
          </li>
        </ul>
      </div>
    </section>
    <section id="start-objective" class="resume-section p-3 p-lg-5 d-flex d-column">
      <div class="aboutme-container mb-4 mb-md-3">
        <div class="aboutme-paragraph lead">
          <p>
            As a lifelong polyglot and software developer, I have come to describe myself as a Language Data Specialist.
          </p>
        </div>
        <div class="aboutme-paragraph lead">
          <p>
            Through living, studying and working internationally, I developed keen and varied insight into human interaction, motivation, and the sheer nitty-gritty of life - witnessed everywhere from hospitals to courthouses and union halls in between.
          </p>
        </div>
        <div class="aboutme-paragraph lead">
          <p>
            Over time, considering where spoken languages and behavior intersect with data got me interested in programming. The call to bridge the gap between my linguistic expertise and (long-dormant) digital dexterity came in the form of my brotherâ€™s coaching, freecodecamp, and finally formal training at Washington University St. Louis Data Analytics Boot Camp.
          </p>
        </div>
        <div class="aboutme-paragraph lead">
          <p>
            Today, I apply my advanced tech skills to analyses that consider where the cultural and the computational collide. This, combined with my fluency for the most widely spoken languages, allows my clients a unique perspective on their field of choice.
          </p>
        </div>
        <div class="aboutme-paragraph lead">
          <p> Showcasing the evolution of these skills in this - <em>work-in-progress</em> - resume site has been a learning process of its own. This site is hosted on <a href="https://shortpoet.azurewebsites.net">Azure</a> with a linked database to dynamically populate different resume formats. Please check out the <a href="/pdf">PDF-friendly</a> version. And coming soon, a guestbook that will fully illustrate the capabilities of a linked data source! ðŸ¤“</p>
        </div>
      </div>
    </section>
    <section id="experience" class="resume-section p-3 p-lg-5 d-flex flex-column">
      <div class="my-auto">
        <h2 class="mb-4 mb-md-3">Experience</h2>
        <div class="resume-item d-flex flex-column mb-2">
          <h2 class="mb-3">software</h2>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Developer</h3>
              <div class="subheading mb-3">Personal Project</div>
              <div>
                <p>â€¢ Escoba Card Game App </p>
              </div>
              <div>
                <p> â€¢ c#, vuejs </p>
              </div>
              <div>
                <p> â€¢ Game manager handles game logic in dotnet backend server. Game sessions and game state managed by backend. UI state managed by vuex store. </p>
              </div>
              <div>
                <p> â€¢ Future... Drag/Drop swipeable UI - Machine learning to create AI </p>
              </div>
              <div>
                <p> â€¢ In collaboration with Ignacio Soriano
                </p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">October 2019 - ongoing</span></div>
          </div>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Developer</h3>
              <div class="subheading mb-3">Personal Project</div>
              <div>
                <p>â€¢ Poet University </p>
              </div>
              <div>
                <p> â€¢ c#, vuejs, IdentityServer </p>
              </div>
              <div>
                <p> â€¢ An extension of Contoso University in microsoft docs. Randomly generated data from lists and some basic logic I coded before learning about the faker library hehe. Has a working implementation of an IdentityServer auth provider. Frontend client app uses IdentityServer as well as Auth0 to provide authentication for authorization experiments on the protected routes of the University. </p>
              </div>
              <div>
                <p> â€¢ Future... More variety with the faker library. Better auth. Cleaner UI. Complete the university environment.
                </p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">March 2020 - ongoing</span></div>
          </div>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Developer</h3>
              <div class="subheading mb-3">Personal Project</div>
              <div>
                <p>â€¢ Web App and Social Integration Dashboard </p>
              </div>
              <div>
                <p> â€¢ c#, node.js, vuejs, Auth0, Gapi, REST APIs </p>
              </div>
              <div>
                <p> â€¢ Authentication to Google API (gmail, sheets, drive, etc), LinkedIn, Facebook and access to my account's data and programattic access to APIs. User jwt for authentication. Use cases include mass unsubscribe, sending emails that are programatically created, integrating Linkedin searches with gmail messages for job search, and many others. Also a way to explore the intricacies of authentication through the use of Auth0 and Google as identity providers. </p>
              </div>
              <div>
                <p> â€¢ Future... More integrations and built-in functionality. Using headless browser to unsubscribe from senders that don't provide a 'mailto' but rather a url.
                </p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">March 2020 - ongoing</span></div>
          </div>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Software Engineer</h3>
              <div class="subheading mb-3">Boeing</div>
              <div>
                <p>â€¢ Built an application to assess the validity of employee timesheet entries. Managers asked for functionality to parse different types of copy/pasted input of time data. A SQL databse function checked against translation tables maintained in collaboration with the business operations team. Data output was displayed using d3 for data visualization to help understand labor costs. </p>
              </div>
              <div>
                <p> â€¢ Learned and fully implemented advanced features of the Vue framework for a variety of data analytics applications and reconstructing legacy Angular products to help transition an aging library of corporate applications. </p>
              </div>
              <div>
                <p> â€¢ Parts and labor tracking system to integrate the builds across teams ans systems. Collaborated with deparatments across the company to pull and collate relevant data enabling wider-reaching analytics. </p>
              </div>
              <div>
                <p> â€¢ Centralized data vizualization dashboard with responsive layout and reusable data viz components. Charts and tables that are dynamically populated to avoid repetitive hard-coding and save on future similar requests from customers. Wrapper for tableau api to better integrate server and dashboard functionality across apps. Use of recursive components to display org structure drilldown. </p>
              </div>
              <div>
                <p> â€¢ Ensured data integrity of daily loading from source systems mostly using c#/dotnet and sql scripts with some python and R developing new ETL architecture for reusable data analytics pipeline. As new data sources come from increased collaboration, creating a workflow enabled efficient operations. </p>
              </div>
              <div>
                <p> â€¢ Team devops guy - deployment of products using Cloud Foundry, Gitlab CI/CD, bash and powershell scripting, and windows server virtual machine. Things break and someone has to fix them. </p>
              </div>
              <div>
                <p> â€¢ Liaising with other teams/departments to collaborate in data flow and analytics. Collaborated with a diverse and distributed team working with a 12-hour time difference with India and delivered results tracking contributions through Gitlab and Jira in a remote-work environment. Being able to communicate changes and action items needed across teams allows for smoother workflow.
                </p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">April 2019 - April 2020</span></div>
          </div>
        </div>
        <div class="resume-item d-flex flex-column mb-2">
          <h2 class="mb-3">language</h2>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Sinologist</h3>
              <div class="subheading mb-3">World Eye Reports</div>
              <div>
                <p>Shanghai, PRC. Translator, Interpreter, media Sales, wrote articles and sold ads for advertorial project based out of Shanghai.</p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">August 2003 - January 2004</span></div>
          </div>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">QA Tester</h3>
              <div class="subheading mb-3">GTL Media</div>
              <div>
                <p>Translator, linguistic tester, translated texts for various media including video games, film, and legal, checked video games for linguistic bugs and translation errors, translation of .xml databases for in-game script</p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">December 2011 - August 2014</span></div>
          </div>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Executive Assistant to the CEO</h3>
              <div class="subheading mb-3">EZI</div>
              <div>
                <p>Manila, Philippines. Participated in managing daily operations in a bilingual office setting, supervised staff, attended developmental meetings, tracked and prepared development projects, made calls to and from suppliers and manufacturers, purchased investment property, created investment portfolios and marketing and business plans, gaining insights into the world of Big Business.</p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">January 2008 - December 2008</span></div>
          </div>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Freelance Translator / Interpreter</h3>
              <div class="subheading mb-3">Various Interpretation Agencies</div>
              <div>
                <p>Medical, dental and media conferences, on-site medical and legal interpretation for depositions, insurance and immigration interviews, simultaneous interpretation of contract negotiations, guided tours, consecutive and simultaneous interpretation for training meetings, document translation of many types, including legal, medical, media localization, and various other language-related assignments, primarily in Spanish and Tagalog, some French.</p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">July 2011 - Present</span></div>
          </div>
        </div>
        <div class="resume-item d-flex flex-column mb-2">
          <h2 class="mb-3">sales</h2>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Product Specialist - Sales</h3>
              <div class="subheading mb-3">Toyota of Hollywood &amp; Hollywood Electrics â€“ Electric Cycle Dealership</div>
              <div>
                <p>New and Pre-owned Conventional and Electric Car and Cycle Sales; Sales made in English, Spanish, Tagalog, French, and Mandarin, Self-directed marketing efforts</p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">July 2010 - February 2017</span></div>
          </div>
        </div>
        <div class="resume-item d-flex flex-column mb-2">
          <h2 class="mb-3">hospitality</h2>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Grill Chef de Partie</h3>
              <div class="subheading mb-3">Many Different Restaurants - mostly a cook, sometimes a bartender</div>
              <div>
                <p>Post Haus Restaurant - St. Moritz, Switzerland stands out as I ran the gamut from helping plan restaurant concept, designed Grill Station, assisted in menu planning and execution, provided back-end data for owners during the length of the contract, offering particular assistance interpreting for members of the multi-cultural staff and management.</p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">August 1998 - August 2018</span></div>
          </div>
          <div class="resume-item d-flex flex-column flex-md-row mb-5">
            <div class="resume-content mr-auto">
              <h3 class="mb-0">Food delivery logistics</h3>
              <div class="subheading mb-3">DoorDash and Grubhub</div>
              <div>
                <p>Operating 2 apps at the same time to maximize earnings, while maintaining customer satisfaction and high ratings, experienced first hand the effects of data collection of market and labor fluctuations and how they determined the logic behind the apps while I was learning to code.</p>
              </div>
            </div>
            <div class="resume-date text-md-right mt-1"><span class="text-primary">April 2017 - April 2019</span></div>
          </div>
        </div>
      </div>
    </section>
    <section id="education" class="resume-section p-3 p-lg-5 d-flex flex-column">
      <div class="my-auto">
        <h2 class="mb-5">Education</h2>
        <div class="resume-item d-flex flex-column flex-md-row mb-5">
          <div class="resume-content mr-auto">
            <h3 class="mb-0">Washington University in St. Louis</h3>
            <div class="subheading mb-3">Data Analytics Certificate</div>
            <div></div>
          </div>
          <div class="resume-date text-md-right mt-1"><span class="text-primary">Python, Javascript, Tableau</span></div>
        </div>
        <div class="resume-item d-flex flex-column flex-md-row mb-5">
          <div class="resume-content mr-auto">
            <h3 class="mb-0">University of Pennsylvania</h3>
            <div class="subheading mb-3">Bachelor of Arts and Sciences</div>
            <div></div>
          </div>
          <div class="resume-date text-md-right mt-1"><span class="text-primary">Major in East Asian Languages and Cultures, Chinese Concentration, relevant coursework in linguistics and anthropology</span></div>
        </div>
        <div class="resume-item d-flex flex-column flex-md-row mb-5">
          <div class="resume-content mr-auto">
            <h3 class="mb-0">Thacher School</h3>
            <div class="subheading mb-3">High School</div>
            <div></div>
          </div>
          <div class="resume-date text-md-right mt-1"><span class="text-primary">Boarding, emphasis on academics, the outdoors, camping, horsemanship, and environmental awareness, Year Abroad in France</span></div>
        </div>
      </div>
    </section>
    <section id="interests" class="resume-section p-3 p-lg-5 d-flex flex-column">
      <div class="my-auto">
        <h2 class="mb-4 mb-md-3">Interests</h2>
        <p class="mb-4">When I'm not knuckle-deep in code, I like to knead bread, dig gardens, vermicompost and cook tasty meals. I do some yoga and pilates to work on the code-bod. I have been known to be insipired by the dramatic muses. Ask for an accent sometime! Being a big fan of perspective, when I started going underwater at age 11, what once was a world of fear turned into a world of wonder. When I can, SCUBA is pretty sweet stuff. Speaking of perspective, here's some shortpoetry...
        </p>
      </div>
      <div class="short-poetry my-auto">
        <div id="poem-container" class="mt-3 mt-md-0 mr-0 text-center"><img src="" alt="poem" class="img-fluid img-profile rounded"></div>
      </div>
    </section>
    <div class="start target vue-portal-target">
      <div class="ripple-container">
        <div class="ripple-mask ripple-mask-hide">
          <div class="ripple-content">
            Check out the other layout by clicking the pdf icon.
          </div>
        </div>
        <div class="icon-halo icon-halo-hide">
          <div class="icon-circle"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="rocket" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" class="button-float-icon svg-inline--fa fa-rocket fa-w-16 fa-2x" style="transform-origin: 0.4375em 0.5625em;">
              <g transform="translate(256 256)" class="">
                <g transform="translate(-32, 32)  scale(1, 1)  rotate(0 0 0)" class="">
                  <path fill="currentColor" d="M505.12019,19.09375c-1.18945-5.53125-6.65819-11-12.207-12.1875C460.716,0,435.507,0,410.40747,0,307.17523,0,245.26909,55.20312,199.05238,128H94.83772c-16.34763.01562-35.55658,11.875-42.88664,26.48438L2.51562,253.29688A28.4,28.4,0,0,0,0,264a24.00867,24.00867,0,0,0,24.00582,24H127.81618l-22.47457,22.46875c-11.36521,11.36133-12.99607,32.25781,0,45.25L156.24582,406.625c11.15623,11.1875,32.15619,13.15625,45.27726,0l22.47457-22.46875V488a24.00867,24.00867,0,0,0,24.00581,24,28.55934,28.55934,0,0,0,10.707-2.51562l98.72834-49.39063c14.62888-7.29687,26.50776-26.5,26.50776-42.85937V312.79688c72.59753-46.3125,128.03493-108.40626,128.03493-211.09376C512.07526,76.5,512.07526,51.29688,505.12019,19.09375ZM384.04033,168A40,40,0,1,1,424.05,128,40.02322,40.02322,0,0,1,384.04033,168Z" transform="translate(-256 -256)" class=""></path>
                </g>
              </g>
            </svg></div>
        </div>
      </div>
    </div>
    <div class="v-portal" style="display: none;"></div>
  </div>
</div>
`;
